packages/form/src/reatomForm.ts

```diff
@@ -1,176 +1,684 @@
+        !isAtom(fields[key])
+      ) {
+        reinitState(value, fields[key] as unknown as FormFields);
+      }
+      else if (isAtom(fields[key])) {
+        fields[key].initState(value);
+      }
+    }
+  };
+
+  const init = action((initState: FormPartialState<T>) => {
+    reinitState(initState, fields);
+  }, `${name}.init`);
+
+	const checkSchemaValidation = action(async (triggerOnlyFor?: Atom) => {
+    if (!schema)
+			throw new Error('Triggering schema validation without schema');
+    
+    const state = fieldsState();
+		const validation = schema['~standard'].validate(state);
+		const result = validation instanceof Promise ? await wrap(validation) : validation;
+
+		if (result.issues?.length) {
+			for (const issue of result.issues) {
+				const field = resolveFieldByPath(issue.path, fields);
+				if (!field || (triggerOnlyFor && triggerOnlyFor !== field))
+					continue;
+
+				field.validation.merge({
+					error: issue.message,
+					meta: undefined,
+					triggered: true,
+					validating: false,
+				});
+
+				if (triggerOnlyFor)
+					break;
+			}
+		}
+
+		return result;
+	}, `${name}.checkSchemaValidation`);
+
+  const submit = action(async () => {
+    for (const field of fieldsList()) {
+      if (!field.validation().triggered)
+        field.validation.trigger();
+    }
+
+    notify();
+
+    if(validation().validating) {
+      await wrap(
+        (async () => {
+          while(validation().validating)
+            await take(validation, `${name}.onSubmit.validationTake`)
+        })()
+      )
+    }
+
+    const error = validation().error;
+    if (error) throw new Error(error);
+
+    let state: any
+
+		if (schema) {
+			const schemaValidationResult = await wrap(checkSchemaValidation());
+			if (!('value' in schemaValidationResult))
+				throw new Error(schemaValidationResult.issues[0]?.message ?? 'Unknown schema error');
+
+			state = schemaValidationResult.value;
+		}
+		else {
+			state = fieldsState();
+		}
+    
+    if (validate) {
+      const promise = validate(state);
+      if (promise instanceof Promise)
+        await wrap(promise);
+    }
+
+    if (onSubmit) {
+      const promise = onSubmit(state);
+      if (promise instanceof Promise)
+        await wrap(promise);
+    }
+
+    submitted(true);
+
+    if (resetOnSubmit)
+      reset();
+  }, `${name}.onSubmit`).extend(
+    withAsync(),
+    withAbort(),
+    target => {
+      const error = createAtom<Error | undefined>({
+        initState: undefined,
+        computed: (state) => {
+          ifCalled(target.onFulfill, () => {
+            state = undefined;
+          })
+          ifCalled(target.onReject, ({ error }) => {
+            state = error;
+          })
+          return state;
+        }
+      }, `${target.name}.error`).actions((target) => ({
+        reset: () => target(undefined)
+      }))
+
+      target.onFulfill.extend(withCallHook(() => error()))
+      target.onReject.extend(withCallHook(() => error()))
```

Comment:
`withAsync` already added `error` atom, what is your case? Fill free to add `reset` action to `withAsync` error sources